R/check_meta_filter_group_match.R

```diff
@@ -0,0 +1,80 @@
+#' Check filter groups match in meta and data
+#'
+#' Ensure all filter groups from the meta data are found in the data file.
+#'
+#' @inheritParams precheck_col_to_rows
+#'
+#' @inherit check_filename_spaces return
+#'
+#' @family check_meta
+#'
+#' @examples
+#' check_meta_filter_group_match(example_data, example_meta)
+#' check_meta_filter_group_match(example_data, example_meta, verbose = TRUE)
+#' @export
+check_meta_filter_group_match <- function(
+  data,
+  meta,
+  verbose = FALSE,
+  stop_on_error = FALSE
+) {
+  meta_filter_groups <- data.frame(
```

Comment:
Just realised that this is doing something a little off as it converts the input data frame into a data frame within a data frame, which then stops the test from working.
```
  meta_filter_groups <- meta |>
    dplyr::filter(
      !(is.na(filter_grouping_column) | filter_grouping_column == "")
    )
```

Noticed this when I ran screen_dfs on a modified example_data and example_meta with the meta adapted to have a redundant filter grouping column and that, so worth double checking that scenario as well as the standalone test data frames.